cpp/models/ide_secir/model.cpp

```diff
@@ -21,101 +21,118 @@
+            m_transitions.get_last_time(),
+            TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count * m_num_agegroups, 0.));
     }
     else {
         // Initialize m_populations with zero as the first point of time if no data is provided for the transitions.
         // This can happen for example in the case of initialization with real data.
         m_populations.add_time_point<Eigen::VectorXd>(
-            0, TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count, 0));
+            0, TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count * m_num_agegroups, 0));
     }
 
     // Set deaths at simulation start time t0.
-    m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Dead)] = deaths;
+    for (int group = 0; group < m_num_agegroups; group++) {
+        int Di                             = get_state_flat_index(Eigen::Index(InfectionState::Dead), group);
+        m_populations[Eigen::Index(0)][Di] = deaths[static_cast<Eigen::Index>((size_t)group)];
+    }
 }
 
 // ---- Functionality to calculate the sizes of the compartments for time t0. ----
-void Model::compute_compartment_from_flows(ScalarType dt, Eigen::Index idx_InfectionState,
+void Model::compute_compartment_from_flows(ScalarType dt, Eigen::Index idx_InfectionState, AgeGroup group,
                                            Eigen::Index idx_IncomingFlow, int idx_TransitionDistribution1,
                                            int idx_TransitionDistribution2)
 {
+
     ScalarType sum       = 0;
     ScalarType calc_time = 0;
     // Determine relevant calculation area and corresponding index.
-    if ((1 - parameters.get<TransitionProbabilities>()[idx_TransitionDistribution1]) > 0) {
-        calc_time =
-            std::max(parameters.get<TransitionDistributions>()[idx_TransitionDistribution1].get_support_max(dt, m_tol),
-                     parameters.get<TransitionDistributions>()[idx_TransitionDistribution2].get_support_max(dt, m_tol));
+    if ((1 - parameters.get<TransitionProbabilities>()[group][idx_TransitionDistribution1]) > 0) {
+        calc_time = std::max(
+            parameters.get<TransitionDistributions>()[group][idx_TransitionDistribution1].get_support_max(dt, m_tol),
+            parameters.get<TransitionDistributions>()[group][idx_TransitionDistribution2].get_support_max(dt, m_tol));
     }
     else {
-        calc_time = parameters.get<TransitionDistributions>()[idx_TransitionDistribution1].get_support_max(dt, m_tol);
+        calc_time =
+            parameters.get<TransitionDistributions>()[group][idx_TransitionDistribution1].get_support_max(dt, m_tol);
     }
 
     Eigen::Index calc_time_index = (Eigen::Index)std::ceil(calc_time / dt) - 1;
 
     Eigen::Index num_time_points = m_transitions.get_num_time_points();
 
+    int flow_index = get_transition_flat_index(idx_IncomingFlow, static_cast<Eigen::Index>((size_t)group));
```

Comment:
Can you add a comment what flow_index and state_index are referring to, i.e. m_transitions and m_populations?